File: problems/leetcode/0001-two-sum/solution.go
package main

func twoSum(nums []int, target int) []int {
	hashMap := make(map[int]int)

	for i := 0; i < len(nums); i++ {
		complement := target - nums[i]
		if idx, exists := hashMap[complement]; exists {
			return []int{idx, i}
		}
	}
	return nil
}

File: problems/leetcode/0001-two-sum/solution.py
def twoSum(nums, target):
    hash_map = {}
    for index, num in enumerate(nums):
        complement = target - num
        if complement in hash_map:
            return [hash_map[complement], index]
        
        hash_map[num] = index
 
      
    return []


print(twoSum([11, 5, 7, 4, 78, 2], 9))